src/main/java/de/tum/in/www1/artemis/service/CourseScoreCalculationService.java

```diff
@@ -98,35 +124,154 @@ public CourseScoresDTO calculateCourseScores(long courseId, Collection<Long> stu
+        for (Exercise exercise : course.getExercises()) {
+            exercise.setCourse(course);
+            StudentParticipation exerciseParticipation;
+            // This method is used in the CourseResource where the course is first fetched with lazy participations, and participations are then fetched separately and added to the
+            // course if found.
+            // If no participations are found for the course, no value is set to the course's participations and trying to access them here would throw a
+            // LazyInitializationException.
+            if (Hibernate.isInitialized(exercise.getStudentParticipations())) {
+                exerciseParticipation = exercise.getStudentParticipations().iterator().next();
+                exerciseParticipation.setExercise(exercise);
+                studentParticipations.add(exerciseParticipation);
+            }
+        }
+
+        // Get scores per exercise type for the course (used in course-statistics.component i.a.).
+        Map<String, CourseScoresDTO> scoresPerExerciseType = calculateCourseScoresPerExerciseType(course, studentParticipations, userId);
+
+        // Get participation results (used in course-statistics.component).
+        List<Result> participationResults = new ArrayList<>();
+        for (StudentParticipation studentParticipation : studentParticipations) {
+            participationResults.add(getResultForParticipation(studentParticipation, studentParticipation.getIndividualDueDate()));
+        }
+
+        return new CourseForDashboardDTO(course, scoresPerExerciseType, participationResults);
+    }
+
+    /**
+     * Prepares all entities required for calculateCourseScoresForStudent and calls it multiple times to retrieve the scores per exercise type for the specified student in the
+     * specified course.
+     * In addition to the scores per exercise type, the total scores per course are calculated.
+     *
+     * @param course the course to calculate the scores for
+     * @param userId the id of the user whose scores will be calculated
+     * @return a map of the scores for the different exercise types (total, for programming exercises etc.). For each type, the map contains the max and reachable max points and
+     *         the scores of the current user.
+     */
+    private Map<String, CourseScoresDTO> calculateCourseScoresPerExerciseType(Course course, List<StudentParticipation> studentParticipations, long userId) {
+
+        Map<String, CourseScoresDTO> scoresPerExerciseType = new HashMap<>();
+
+        // Retrieve required entities
+
+        Set<Exercise> courseExercises = course.getExercises();
+
+        MaxAndReachablePoints maxAndReachablePoints = calculateMaxAndReachablePoints(courseExercises);
+
+        List<PlagiarismCase> plagiarismCases = new ArrayList<>();
+        for (Exercise exercise : courseExercises) {
+            if (Hibernate.isInitialized(exercise.getPlagiarismCases())) {
```

Comment:
Same here, shouldn't we load them if they are not initialized?

Reply:
See comment above.